modules/telemetry/telemetry.php

```diff
@@ -0,0 +1,232 @@
+	 * @param string $term_name The term's name
+	 * @param string $slug The term's slug
+	 */
+
+	public static function record_add_custom_status( int $term_id, string $term_name, string $term_slug ): void {
+		self::$tracks->record_event( 'custom_status_created', [
+			'term_id' => $term_id,
+			'name'    => $term_name,
+			'slug'    => $term_slug,
+		] );
+	}
+
+	/**
+	 * Record an event when a custom status is deleted
+	 *
+	 * @param int $term_id The custom status term ID
+	 * @param string $slug The status slug
+	 */
+	public static function record_delete_custom_status( int $term_id, string $slug ): void {
+		self::$tracks->record_event( 'custom_status_deleted', [
+			'term_id' => $term_id,
+			'slug'    => $slug,
+		] );
+	}
+
+	/**
+	 * Record an event when a custom status is updated
+	 *
+	 * @param WP_Term $updated_status The updated status WP_Term object.
+	 * @param array $update_args The arguments used to update the status.
+	 */
+	public static function record_update_custom_status( WP_Term $updated_status, array $update_args ): void {
+		$is_position_update = 1 === count( $update_args ) && isset( $update_args['position'] );
+		if ( $is_position_update ) {
+			// Ignore position changes, as they fire for every custom status when statuses are reordered
+			return;
+		}
+
+		self::$tracks->record_event( 'custom_status_changed', [
+			'term_id' => $updated_status->term_id,
+			'name'    => $updated_status->name,
+			'slug'    => $updated_status->slug,
+		] );
+	}
+
+	// Notification events
+
+	/**
+	 * Record an event when a notification is sent
+	 *
+	 * @param int $post_id The post ID of the post that was updated.
+	 * @param bool $is_email_scheduled True if an email was scheduled as part of the notification, false otherwise.
+	 * @param bool $is_webhook_scheduled True if a webhook was scheduled as part of the notification, false otherwise.
+	 */
+	public static function record_notification_sent( int $post_id, $is_email_scheduled, $is_webhook_scheduled ): void {
+		self::$tracks->record_event( 'notification_sent', [
+			'post_id' => $post_id,
+			'email'   => $is_email_scheduled,
+			'webhook' => $is_webhook_scheduled,
+		] );
+	}
+
+	// Settings events
+
+	/**
+	 * Record an event when the plugin is upgraded
+	 *
+	 * @param string $previous_version The previous version
+	 * @param string $new_version The new version
+	 */
+	public static function record_admin_update( string $previous_version, string $new_version ): void {
+		$custom_statuses      = CustomStatus::get_custom_statuses();
+		$supported_post_types = HelperUtilities::get_supported_post_types();
+
+		$custom_status_posts = 0;
+
+		foreach ( $supported_post_types as $post_type ) {
+			// Get all posts count for this post type
+			$posts_count = wp_count_posts( $post_type );
+
+			foreach ( $custom_statuses as $status ) {
+				if ( isset( $posts_count->{ $status->slug } ) ) {
+					$custom_status_posts += (int) $posts_count->{ $status->slug };
```

Comment:
This event has been modified to only count posts in a custom status. Previously we were also tracking published posts, but I didn't see the need for that datapoint.

Reply:
I think we want "published" as well for a few reasons:

* Since each of these events has a timestamp, it might show associations in workflow shapes, complications, and publishing velocity.
* If we have a site with zero published posts, we know something is wonky.
* If we started to record Post IDs we could track the time it takes for a post to go from start to finish.

Maybe this is low value/far out, but I'm interested in others thoughts.

Reply:
I think that's a good reason, I'll add it back in.

Reply:
Agreed on that. This would also help us better support complicated workflows, and would be beneficial down the line to target new customers who haven't used WordPress before as they are fond of Adobe, etc instead.